espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Handlers/SendKeys.java

```diff
@@ -3,71 +3,34 @@
 import android.support.test.espresso.PerformException;
 import android.support.test.espresso.ViewInteraction;
 
-import com.google.gson.Gson;
-
-import java.io.IOException;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.UUID;
-
-import fi.iki.elonen.NanoHTTPD;
-import io.appium.espressoserver.lib.Http.Response.AppiumResponse;
-import io.appium.espressoserver.lib.Http.Response.BadRequestResponse;
-import io.appium.espressoserver.lib.Http.Response.BaseResponse;
-import io.appium.espressoserver.lib.Http.Response.InternalErrorResponse;
-import io.appium.espressoserver.lib.Http.Response.InvalidSessionResponse;
-import io.appium.espressoserver.lib.Model.AppiumStatus;
+import io.appium.espressoserver.lib.Handlers.Exceptions.BadParametersException;
 import io.appium.espressoserver.lib.Model.Element;
-import io.appium.espressoserver.lib.Model.Session;
+import io.appium.espressoserver.lib.Model.TextParams;
 
 import static android.support.test.espresso.action.ViewActions.typeText;
 
 
-public class SendKeys implements RequestHandler {
-
-    public BaseResponse handle(NanoHTTPD.IHTTPSession session, Map<String, String> uriParams) {
-        // If the SessionID is invalid, return InvalidSessionResponse
-        // TODO: Fix SessionID handling redundancies
-        if (!uriParams.get("sessionId").equals(Session.getGlobalSessionId())) {
-            return new InvalidSessionResponse(uriParams.get("sessionId"));
-        }
-
-        String id = uriParams.get("elementId");
-        ViewInteraction viewInteraction = Element.getCache().get(id);
+public class SendKeys implements RequestHandler<TextParams> {
 
-        // NanoHTTP requires call to parse body before we can get the parameters
-        // TODO: Move parameter parsing into Router.java
-        Map<String, List<String>> parameters;
-        try {
-            Map<String, String> files = new HashMap<>();
-            session.parseBody(files);
+    public Object handle(TextParams params) {
+        String id = params.getElementId();
+        ViewInteraction viewInteraction = Element.getById(id);
 
-            Gson gson = new Gson();
-            parameters = gson.fromJson(files.get("postData"), Map.class);
-        } catch (NanoHTTPD.ResponseException e) {
-            return new BadRequestResponse("Could not parse parameters");
-        } catch (IOException e) {
-            return new InternalErrorResponse("Internal server error has occurred");
+        // Convert the array of text to a String
+        String[] textArray = params.getValue();
+        StringBuilder stringBuilder = new StringBuilder();
+        for (String text: textArray) {
```

Comment:
One can also use StringUtils.join from http://commons.apache.org/proper/commons-lang/ for this purpose